include/options/registry.php

```diff
@@ -0,0 +1,133 @@
+		),
+		'hide_default'        => array(
+			'class'       => Business\Hide_Default::class,
+			'default'     => true,
+			'description' => 'Remove language code in URL for default language: true to hide, false to display.',
+		),
+		'rewrite'             => array(
+			'class'       => Primitive\Boolean::class,
+			'default'     => true,
+			'description' => 'Remove /language/ in pretty permalinks: true to remove, false to keep.',
+		),
+		'redirect_lang'       => array(
+			'class'       => Primitive\Boolean::class,
+			'default'     => false,
+			'description' => 'Remove the page name or page id from the URL of the front page: true to remove, false to keep.',
+		),
+		// Detect browser language.
+		'browser'             => array(
+			'class'       => Business\Browser::class,
+			'default'     => false,
+			'description' => 'Detect browser language on front page: true to detect, false to not detect.',
+		),
+		// Media.
+		'media_support'       => array(
+			'class'       => Primitive\Boolean::class,
+			'default'     => false,
+			'description' => 'Translate media: true to translate, false to not translate.',
+		),
+		// Custom post types and taxonomies.
+		'post_types'          => array(
+			'class'       => Business\Post_Types::class,
+			'default'     => array(),
+			'description' => 'List of post types to translate.',
+			'args'        => array( 'string' ),
+		),
+		'taxonomies'          => array(
+			'class'       => Business\Taxonomies::class,
+			'default'     => array(),
+			'description' => 'List of taxonomies to translate.',
+			'args'        => array( 'string' ),
+		),
+		// Synchronization.
+		'sync'                => array(
+			'class'       => Business\Sync::class,
+			'default'     => array(),
+			'description' => 'List of data to synchronize.',
+			'args'        => array( 'string' ),
+		),
+		// Internal.
+		'default_lang'        => array(
+			'class'       => Business\Language_Slug::class,
+			'default'     => '',
+			'description' => 'Slug of the default language.',
+		),
+		'nav_menus'           => array(
+			'class'       => Business\Nav_Menu::class,
+			'default'     => array(),
+			'description' => 'Translated navigation menus for each theme.',
+		),
+		'language_taxonomies' => array(
+			'class'       => Business\Language_Taxonomies::class,
+			'default'     => array(),
+			'description' => 'List of language taxonomies used for custom DB tables.',
+			'args'        => array( 'string' ),
+		),
+		// Read only.
+		'first_activation'    => array(
+			'class'       => Primitive\Integer::class,
+			'default'     => array(),
+			'description' => 'Time of first activation of Polylang.',
+		),
+		'previous_version'    => array(
+			'class'       => Primitive\String_Type::class,
+			'default'     => '',
+			'description' => "Polylang's previous version.",
+		),
+		'version'             => array(
+			'class'       => Primitive\String_Type::class,
+			'default'     => POLYLANG_VERSION,
```

Comment:
Should constant definition be checked before for extra safety? Or is it overkill?

Reply:
This constant is literally the first thing that Polylang defines, so I can't see any scenario where this class can be called before the constant is defined. Do you have one in mind?

Reply:
I see, thanks.
Maybe in a CLI environment (not implemented yet) or in unit tests. But we should be able to work around that.